### ISSUE TITLE
Proposal: Add support for custom base image repository in the Dockerfile

### ISSUE THREAD

**Issue by matiasAS** [dokku/github-action#86]
Description of the problem
I have been encountering rate limit errors when pulling images from Docker Hub. To address this, I'm using a Harbor project as a cache. It would be helpful to modify the Dockerfile in the GitHub Action to allow customizing the base image repository.

Proposal
Modify the Dockerfile to allow the use of a custom repository for the FROM statement, like this:

`
ARG REPOSITORY
FROM ${REPOSITORY}/dokku/ci-docker-image:0.13.0
`

Add a new optional parameter, REPOSITORY, to the GitHub Action.
- If the REPOSITORY variable is provided, it should be used in the FROM statement.
- If it’s empty, it should default to:

`
FROM dokku/ci-docker-image:0.13.0
`

Expected Results
- Users can configure the REPOSITORY in the GitHub Action inputs.
- If the REPOSITORY is not set, the action should default to using dokku/ci-docker-image:0.13.0 from Docker Hub.

@josegonzalez i from Chile, 

Regards
Saludos

**Comment by josegonzalez** [comment]
I don't think this is a possibility with github actions. Do you know of another composite action that does what you are asking for?

**Comment by matiasAS** [comment]
I forked the repository, modified the dockerfile, and then in actions changed it to my repository

@josegonzalez  Is there a possibility of speaking more directly in Spanish?

**Comment by josegonzalez** [comment]
Feel free to post in spanish and I'll try and respond.

**Comment by matiasAS** [comment]
gracias @josegonzalez 
Intentaré explicarlo mejor
Uso dokku hace casi 1 año
Despliego con github actions , tengo mi cluster k0s y ahi actions runner controller para runners self hosted.

El error de la imagen ya lo solucioné.
Trabajo solo, necesito tener los menores costos posibles, y puede que contratar Docker pro sea la solución más rapida, pero obté por probar harbor.
La idea del proyecto cache de harbor es que al hacer pull de la imagen, docker primero va a harbor y si no la encuentra a docker hub y hace una copia en harbor para que la proxima vez la obtenga de harbor. Pero para que esto funcione es necesario adelante de la imagen agregar el repositorio docker de harbor. Por eso que el dockerfile en vez de ser

FROM dokku/ci-docker-image:0.13.1
deberia ser
FROM <repositorio-harbor>/dokku/ci-docker-image:0.13.1

Con el fork del repositorio, y referencandolo en el workflow de github actions, funciona, pero lo que decia es que en la confuguracion de la action
dokku/github-action@master
esté el parametro de configurar ese repositorio en el dockerfile para no usar la opción forkeada y perder las actualizaciones

**Comment by josegonzalez** [comment]
Tienes un ejemplo de otro composite github action que usa un parametro para el docker image?

**Comment by matiasAS** [comment]
Nunca he construido ni manipulado un gihub action desde que empezé a usarlo, pocos meses, en lo primero que pienso es un build ardg porque en el workflow se hace build de la imagen, entonces ahi seria pasarle el parametro

**Comment by josegonzalez** [comment]
No se puede utilizar build args en el workflow. El documentacion esta acqui: https://docs.github.com/en/actions/sharing-automations/creating-actions/creating-a-docker-container-action

Lo que quieres hacer no es posible en github actions.

**Comment by matiasAS** [comment]
probaré igual en el fork

# action.yml
name: 'Hello World'
description: 'Greet someone and record the time'
inputs:
  who-to-greet:  # id of input
    description: 'Who to greet'
    required: true
    default: 'World'
outputs:
  time: # id of output
    description: 'The time we greeted you'
runs:
  using: 'docker'
  image: 'Dockerfile'
  args:
    - ${{ inputs.who-to-greet }}

**Comment by matiasAS** [comment]
Hola @josegonzalez , encontré una step que hace lo que queria que hiciera esta step!!, 
https://github.com/dokku/github-action

en los with, parametro opcional "image", y en el workflow se hace pull desde ahi.

Saludos

**Comment by matiasAS** [comment]
@josegonzalez lo que hice funciona sin problemas!, puedo especificar mi imagen apuntando a harbor como queria!!!

**Comment by josegonzalez** [comment]
Puedes hacer un PR?

**Comment by matiasAS** [comment]
@josegonzalez si claro, no hay problema

**Comment by matiasAS** [comment]
@josegonzalez listo!
